### ISSUE TITLE
Pull broken macOS 1.7 builds

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/numpy-feedstock#57]
Could we please pull the broken macOS builds of NumPy 1.7? They are not linked to OpenBLAS and are causing issues. Thanks.

**Comment by ocefpaf** [comment]
What cause of issue? They are not broken and work perfectly. Anyways you have access and can remove them if you really want.

**Comment by ocefpaf** [comment]
If you are going to re-open at least be specific on the "issues."

**Comment by jakirkham** [comment]
The are linked to the wrong BLAS, which will cause issues with things that do try to link with OpenBLAS (e.g. crashes). It seems you are already seeing this with SciPy.

I'm open to discussion, but not having my issue slammed shut. Please be respectful. Thanks.

**Comment by ocefpaf** [comment]
>  It seems you are already seeing this with SciPy.

No. See the actual error on SciPy. They are all using np19 with openblas. np17 is a accelerate alternative not used anywhere.

**Comment by ocefpaf** [comment]
> I'm open to discussion, but not having my issue slammed shut.

You are not slammed shut. You are asked to be specific instead of saying "issues."

**Comment by jakirkham** [comment]
Look these build are claiming they are using OpenBLAS and they are not (they are using Accelerate). This is a problem. At a bare minimum it is false advertising, in the worst case we will end up with a mix of BLASes, which will cause issues.

**Comment by jakirkham** [comment]
By mix of BLASes, I mean users will attempt to use OpenBLAS when this is not. If they link to NumPy there will be duplicate symbols, which will cause a range of issues.

**Comment by ocefpaf** [comment]
> Look these build are claiming they are using OpenBLAS 

Copy the lines here with that information for completeness.

**Comment by ocefpaf** [comment]
See https://travis-ci.org/conda-forge/scipy-feedstock/jobs/257774515#L224 and https://travis-ci.org/conda-forge/numpy-feedstock/jobs/257746433

```
+ conda inspect linkages -p /Users/travis/miniconda3/conda-bld/numpy_1501079622353/_t_env numpy

numpy

-----



openblas-0.2.20-1:

    libopenblas.dylib (lib/libopenblas.dylib)



system:

    libSystem.B.dylib (/usr/lib/libSystem.B.dylib)

    libc++.1.dylib (/usr/lib/libc++.1.dylib)
```

If you have evidence of the contrary post the logs.

**Comment by jakirkham** [comment]
That PR is using NumPy 1.9. Ok, given it is the latest SciPy version and using the NumPy 1.9 build that is ok, I'm not worried about that case.

It is the NumPy 1.7 build that does worry me. Here are the [lines]( https://travis-ci.org/conda-forge/numpy-feedstock/builds/257818810#L3993-L3996 ) where I'm seeing NumPy 1.7 using Accelerate for reference. Not that I think this is in dispute, just including them for completeness.

**Comment by ocefpaf** [comment]
Basically you repeated what I said above but ignoring the fact that nothings that uses numpy and openblas will use np17 :unamused:

**Comment by jakirkham** [comment]
Though the doc proposal is still suggesting we use NumPy 1.7 across the board. We are still packaging a NumPy 1.7 version that links to the wrong BLAS. These seem fundamentally problematic even if things like SciPy are built with NumPy 1.9. It is still possible for a user to pick NumPy 1.7 when they are linking to a BLAS directly and get issues, no?

**Comment by ocefpaf** [comment]
> Though the doc proposal is still suggesting we use NumPy 1.7 across the board.

People are supposed to know the minimum numpy for their builds. If they don't the build will break in the CI anyways. I really don't see the problem. Can you find a single feedstock where that may happen?

**Comment by jakirkham** [comment]
How are they suppose to know this if we don't either completely exclude NumPy 1.7 and update the docs, or note in the docs that if they use a BLAS they need to set different Python minimums?

Sure multiple examples `scipy`, `scikit-learn`, `opencv`, etc. These all use `python`, `numpy`, and a BLAS. Not to mention there are cases where a lower-level C++ library uses a BLAS (@scopatz has some of these) is later linked into a higher-level separate Python package. These also need this NumPy 1.9 constraint, but are harder to identify. There are almost certainly more given how many packages are in `conda-forge` and I don't think we as core can police them effectively. So some kind of doc update is needed to help people understand how to use `python`, `numpy`, and BLAS together. Ideally we update to NumPy 1.9, which I think we both agree on, but barring that we need to spell out that NumPy 1.7 is broken with respect to using the correct BLAS.

I did. I see no mention of needing to set `numpy >=1.9  # [py27]` when a BLAS is present. Could you please indicate the line if it is there?

**Comment by ocefpaf** [comment]
> How are they suppose to know this if we don't either completely exclude NumPy 1.7

I really don't get that sentence. They should know that the package requires and no package that links to openblas requires 1.7.

> Sure multiple examples `scipy`, `scikit-learn`, `opencv`, etc. These all use `python`, `numpy`, and a BLAS. Not to mention there are cases where a lower-level C++ library uses a BLAS (@scopatz has some of these) is later linked into a higher-level separate Python package. These also need this NumPy 1.9 constraint, but are harder to identify. There are almost certainly more given how many packages are in `conda-forge` and I don't think we as core can police them effectively. So some kind of doc update is needed to help people understand how to use `python`, `numpy`, and BLAS together. Ideally we update to NumPy 1.9, which I think we both agree on, but barring that we need to spell out that NumPy 1.7 is broken with respect to using the correct BLAS.

I checked a few of those and they all need np>17. Anyways, it is pointless to discuss this with you as you pointed to a non-existing issue with scipy and later you tried to create an issue.  Go ahead and remove them if you want. Like I said you are only depriving some people of an alternative.

**Comment by isuruf** [comment]
@ocefpaf, the blas features are supposed to make sure that there is only one type of blas in an environment. In numpy 1.7, the `blas_openblas` feature is set, but it is using a different blas. I'd like to see numpy 1.7 pinning to something other than `blas 1.1 openblas` on osx.

**Comment by ocefpaf** [comment]
Yep. The right fix would be to add the proper blas metadata to those packages. I did not do that already b/c they are harmless at the moment. Still, like I mentioned above, I am not against removing them. I just re-iterate that the changes of a package that uses np17 and links to openblas is quite low.

Also, by removing them we'll rely on `mkl` np17 from defaults and the same "virtual" problem will exists.

TL;DR the proper fix is what @isuruf suggested above. Still, there won't be package harmed by this.

However, the real important question to answer is: https://github.com/conda-forge/conda-forge.github.io/pull/415#issuecomment-317053353 (Made 6 days ago.)

**Comment by jakirkham** [comment]
> Like I mentioned scipy is limited to np19, but you are insisting in ignoring it.

`scikit-learn` still can't pin `numpy` 1.7 for `python` 2.7 during the build regardless of what fix `scipy` incorporates. `opencv` doesn't require `scipy` to build or run. So fixing `scipy` has no effect on `opencv` and it will need the same hack `scipy` uses. This also has no effect on the aforementioned Python-bound C++ BLAS using library problem, which again `scipy` has no effect on. In short, I fail to see how `scipy` fixes the problem for any of these. Feel free to elaborate though.

**Comment by ocefpaf** [comment]
> scikit-learn still can't pin numpy 1.7

Of course it cannot. It uses np18 minimum. What are you talking about? Do you think we are forcing np17? Please read the thread carefully to understand how this works. You are very confused.

**Comment by jakirkham** [comment]
The documentation proposal doesn't say that we can use any NumPy we want for Python 2.7. It says, ["you need to build against the oldest possible version of numpy that is forwards compatible"]( https://github.com/conda-forge/conda-forge.github.io/pull/415/files#diff-29e8b4bf4d69b118f7b3c877f2f39478R76 ) and then spells out this is NumPy 1.7 for Python 2.7. As noted before, we need to at a minimum make clear that users can and, in some cases, should use newer versions. One case is BLAS and another, as you have noted, is if the package doesn't support that version. Though I must admit that if we start allowing people to use newer versions, it does call into question the attempt to even support such old NumPy versions.

**Comment by ocefpaf** [comment]
The documentation proposal is a suggestion. Like I said many times there is no package that can be built with numpy 1.7 and will link to openblas.  Anyways it is really pointless try to explain something to you when you don't really know what you want.

1. The issue you mention about scipy does not exist.
2. The packages are harmless but you can remove if you want like I already said.
3. If you think the docs are "forcing" people to do that feel free to modify it.

**Comment by jakirkham** [comment]
Pulled the macOS 1.7 packages.
